p4testgen fails to compile on recent Fedora Linux versions

I am giving the symptom in the description.  I do not know what the root cause is, but it might be something like "more recent versions of G++ used on recent Fedora Linux versions are newer than recent versions of Ubuntu Linux, and there are new warnings or errors caught by those newer G++ versions in p4testgen source code".  That is only a guess, though, as I am not familiar with the error messages I am getting (more below).

More detailed symptoms:

Attempting to run these commands on a freshly installed Fedora 36 Linux system:
```
git clone https://github.com/jafingerhut/p4-guide
cd p4-guide
git checkout fedora-try-install-p4testgen-in-install-p4dev-v7
cd ..
./p4-guide/bin/install-p4dev-v7.sh |& tee log.txt
```
(Note, getting to the point of trying to compile p4testgen code can take about 3 hours with these steps)

gives me the following errors near the end of the output:
```
[ 75%] Building CXX object backends/p4tools/modules/testgen/CMakeFiles/testgen.dir/targets/bmv2/bmv2.cpp.o
[ 76%] Building CXX object backends/p4tools/modules/testgen/CMakeFiles/testgen.dir/targets/bmv2/cmd_stepper.cpp.o
[ 76%] Building CXX object backends/p4tools/modules/testgen/CMakeFiles/testgen.dir/targets/bmv2/concolic.cpp.o
[ 76%] Building CXX object backends/p4tools/modules/testgen/CMakeFiles/testgen.dir/targets/bmv2/constants.cpp.o
[ 76%] Building CXX object backends/p4tools/modules/testgen/CMakeFiles/testgen.dir/targets/bmv2/contrib/bmv2_hash/calculations.cpp.o
[ 76%] Building CXX object backends/p4tools/modules/testgen/CMakeFiles/testgen.dir/targets/bmv2/expr_stepper.cpp.o
/home/andy/p4c/backends/p4tools/modules/testgen/targets/bmv2/expr_stepper.cpp: In lambda function:
/home/andy/p4c/backends/p4tools/modules/testgen/targets/bmv2/expr_stepper.cpp:883:89: error: could not convert ‘P4Tools::P4Testgen::Bmv2::BMv2Constants::GREEN’ from ‘P4Tools::P4Testgen::Bmv2::BMv2Constants::METER_COLOR’ to ‘big_int’ {aka ‘boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<> >’}
  883 |                                                             BMv2Constants::METER_COLOR::GREEN));
      |                                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
      |                                                                                         |
      |                                                                                         P4Tools::P4Testgen::Bmv2::BMv2Constants::METER_COLOR
/home/andy/p4c/backends/p4tools/modules/testgen/targets/bmv2/expr_stepper.cpp: In lambda function:
/home/andy/p4c/backends/p4tools/modules/testgen/targets/bmv2/expr_stepper.cpp:1005:89: error: could not convert ‘P4Tools::P4Testgen::Bmv2::BMv2Constants::GREEN’ from ‘P4Tools::P4Testgen::Bmv2::BMv2Constants::METER_COLOR’ to ‘big_int’ {aka ‘boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<> >’}
 1005 |                                                             BMv2Constants::METER_COLOR::GREEN));
      |                                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
      |                                                                                         |
      |                                                                                         P4Tools::P4Testgen::Bmv2::BMv2Constants::METER_COLOR
make[2]: *** [backends/p4tools/modules/testgen/CMakeFiles/testgen.dir/build.make:580: backends/p4tools/modules/testgen/CMakeFiles/testgen.dir/targets/bmv2/expr_stepper.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2977: backends/p4tools/modules/testgen/CMakeFiles/testgen.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```

If you are curious, you can see the default version of various build tools installed on several recent versions of Ubuntu and Fedora Linux here: https://docs.google.com/spreadsheets/d/1uPde_2sLvX-Ek-Xj_ZaC3NrOa-CZ7_n-Hcox_dQ_VCc/edit#gid=0

Fedora Linux tends to have more recent versions of them, for the same release date.

--- a/backends/p4tools/modules/testgen/targets/bmv2/expr_stepper.cpp
+++ b/backends/p4tools/modules/testgen/targets/bmv2/expr_stepper.cpp
@@ -880,7 +880,7 @@ void Bmv2V1ModelExprStepper::evalExternMethodCall(const IR::MethodCallExpression
                      testBackend);
                  auto &nextState = state.clone();
                  nextState.set(meterResult, IR::getConstant(meterResult->type,
-                                                            BMv2Constants::METER_COLOR::GREEN));
+                                                            static_cast<big_int>(BMv2Constants::METER_COLOR::GREEN)));
                  nextState.popBody();
                  result->emplace_back(nextState);
                  return;
@@ -1002,7 +1002,7 @@ void Bmv2V1ModelExprStepper::evalExternMethodCall(const IR::MethodCallExpression
                      "(GREEN).",
                      testBackend);
                  nextState.set(meterResult, IR::getConstant(meterResult->type,
-                                                            BMv2Constants::METER_COLOR::GREEN));
+                                                            static_cast<big_int>(BMv2Constants::METER_COLOR::GREEN)));
                  nextState.popBody();
                  result->emplace_back(nextState);
                  return;

--- a/backends/p4tools/modules/testgen/targets/bmv2/expr_stepper.cpp
+++ b/backends/p4tools/modules/testgen/targets/bmv2/expr_stepper.cpp
@@ -879,8 +879,10 @@ void Bmv2V1ModelExprStepper::evalExternMethodCall(const IR::MethodCallExpression
                      "meter.execute_meter not implemented for %1%. Choosing default value (GREEN).",
                      testBackend);
                  auto &nextState = state.clone();
-                 nextState.set(meterResult, IR::getConstant(meterResult->type,
-                                                            static_cast<big_int>(BMv2Constants::METER_COLOR::GREEN)));
+                 nextState.set(
+                     meterResult,
+                     IR::getConstant(meterResult->type,
+                                     static_cast<big_int>(BMv2Constants::METER_COLOR::GREEN)));
                  nextState.popBody();
                  result->emplace_back(nextState);
                  return;
@@ -1001,8 +1003,10 @@ void Bmv2V1ModelExprStepper::evalExternMethodCall(const IR::MethodCallExpression
                      "direct_meter.read configuration not possible for %1%. Choosing default value "
                      "(GREEN).",
                      testBackend);
-                 nextState.set(meterResult, IR::getConstant(meterResult->type,
-                                                            static_cast<big_int>(BMv2Constants::METER_COLOR::GREEN)));
+                 nextState.set(
+                     meterResult,
+                     IR::getConstant(meterResult->type,
+                                     static_cast<big_int>(BMv2Constants::METER_COLOR::GREEN)));
                  nextState.popBody();
                  result->emplace_back(nextState);
                  return;

